pandas/core/generic.py

```diff
@@ -4164,12 +4169,16 @@ def resample(self, rule, how=None, axis=0, fill_method=None, closed=None,
         """
         from pandas.tseries.resample import (resample,
                                              _maybe_process_deprecations)
+        if is_list_like(on):
+            raise ValueError("Only a single column may be passed to on")
+        if is_list_like(level):
```

Comment:
actually I think might be able to remove these entirely. When ``TimeGrouper._set_grouper`` get's called, these are validated (same as in groupby). e.g. other fixes, #13907 should work for this as well.